core/modules/replica/Csv.h

```diff
@@ -0,0 +1,189 @@
+/*
+ * LSST Data Management System
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef LSST_QSERV_REPLICA_CSV_H
+#define LSST_QSERV_REPLICA_CSV_H
+
+// System headers
+#include <functional>
+#include <memory>
+
+// Qserv headers
+#include "replica/Common.h"
+
+// This header declarations
+namespace lsst {
+namespace qserv {
+namespace replica {
+namespace csv {
+
+/// The maximum number of characters (including the terminator character) in a row.
+constexpr size_t const MAX_ROW_LENGTH = 64 * 1024;
```

Comment:
I will increase this limit up to 16MB, just in case. A reason why I put 64KB is not that "I don't think someone will ever need more than 640KB of RAM" (IBM PC designers), it's because of the CSV/TSV parser (`sph-partition`) that has a similar limit defined in macro `MAX_LINE_SIZE `. See https://github.com/lsst/partition/blob/master/include/lsst/partition/Constants.h